spec/index.bs

```diff
@@ -2282,23 +2269,59 @@ Issue: Define what it means to be a <dfn dfn>compatible texture format</dfn> wit
-        - If |texture|.{{GPUTexture/[[dimension]]}} is {{GPUTextureDimension/"1d"}},
-            set |resolved|.{{GPUTextureViewDescriptor/dimension}} to {{GPUTextureViewDimension/"1d"}}.
-        - If |texture|.{{GPUTexture/[[dimension]]}} is {{GPUTextureDimension/"2d"}}:
-            - If |texture|.{{GPUTexture/[[textureSize]]}}.[=Extent3D/depth=] is greater than `1`
-                and |resolved|.{{GPUTextureViewDescriptor/arrayLayerCount}} is greater than `1`,
-                set |resolved|.{{GPUTextureViewDescriptor/dimension}} to {{GPUTextureViewDimension/"2d-array"}}.
-            - Otherwise, set |resolved|.{{GPUTextureViewDescriptor/dimension}} to {{GPUTextureViewDimension/"2d"}}.
-        - If |texture|.{{GPUTexture/[[dimension]]}} is {{GPUTextureDimension/"3d"}},
-            set |resolved|.{{GPUTextureViewDescriptor/dimension}} to {{GPUTextureViewDimension/"3d"}}.
+    1. If |resolved|.{{GPUTextureViewDescriptor/dimension}} is `undefined` and
+        |texture|.{{GPUTexture/[[dimension]]}} is:
+        <dl class="switch">
+            : {{GPUTextureDimension/"1d"}}
+            :: Set |resolved|.{{GPUTextureViewDescriptor/dimension}} to {{GPUTextureViewDimension/"1d"}}.
+
+            : {{GPUTextureDimension/"2d"}}
+            :: Set |resolved|.{{GPUTextureViewDescriptor/dimension}} to {{GPUTextureViewDimension/"2d-array"}}
+                if |texture|.{{GPUTexture/[[textureSize]]}}.[=Extent3D/depth=] is greater than `1`
+                and |resolved|.{{GPUTextureViewDescriptor/arrayLayerCount}} is greater than `1`
+                or `undefined`, otherwise set |resolved|.{{GPUTextureViewDescriptor/dimension}} to
+                {{GPUTextureViewDimension/"2d"}}.
+
+            : {{GPUTextureDimension/"3d"}}
+            :: Set |resolved|.{{GPUTextureViewDescriptor/dimension}} to {{GPUTextureViewDimension/"3d"}}.
+        </dl>
+    1. If |resolved|.{{GPUTextureViewDescriptor/arrayLayerCount}} is `undefined` and
+        |resolved|.{{GPUTextureViewDescriptor/dimension}} is:
+        <dl class="switch">
+            : {{GPUTextureDimension/"1d"}}, {{GPUTextureDimension/"2d"}}, or {{GPUTextureDimension/"3d"}}
+            :: Set |resolved|.{{GPUTextureViewDescriptor/arrayLayerCount}} to `1`.
+
+            : {{GPUTextureViewDimension/"cube"}}
+            :: Set |resolved|.{{GPUTextureViewDescriptor/arrayLayerCount}} to `6`.
+
+            : {{GPUTextureViewDimension/"2d-array"}}
+            :: Set |resolved|.{{GPUTextureViewDescriptor/arrayLayerCount}} to
+                |texture|.{{GPUTexture/[[textureSize]]}}.[=Extent3D/depth=] &minus;
+                {{GPUTextureViewDescriptor/baseArrayLayer}}.
+
+            : {{GPUTextureViewDimension/"cube-array"}}
+            :: Set |resolved|.{{GPUTextureViewDescriptor/arrayLayerCount}} to
+                ((|texture|.{{GPUTexture/[[textureSize]]}}.[=Extent3D/depth=] &minus;
+                {{GPUTextureViewDescriptor/baseArrayLayer}}) &divide; `6`) &times; `6`.
+        </dl>
```

Comment:
I'm not sure quite this level of magic is desired in the defaulting, we should probably chat about it.

Though simply defaulting to `array layer count of texture - baseArrayLayer` only really makes sense for 2d-array, I thought more about the defaulting (melted my brain for a while) and maybe it's best to be as straightforward as possible. Thought (future-proofed against possible 1d-array, subtly flips\* the original logic prior to #1406):

- If resolved.arrayLayerCount is undefined:
    - If descriptor.dimension is "1d", "2d", or "3d", set resolved.arrayLayerCount to 1.
      Otherwise, set resolved.arrayLayerCount to (texture layer count - descriptor.baseArrayLayer).
      (Note this could result in an invalid descriptor if the dimension is cube or cube-array.)
- If resolved.dimension is undefined:
    - If texture.dimension is 1d: it is assumed texture layer count is 1; set resolved.dimension to 1d.
    - If texture.dimension is 2d and texture layer count is 1: set resolved.dimension to 2d.
    - If texture.dimension is 2d and texture layer count > 1: set resolved.dimension to 2d-array.
    - If texture.dimension is 3d: it is assumed texture layer count is 1; set resolved.dimension to 3d.

\*In particular now if you want a `"2d"` view on a `"2d"` texture with depth>1, you now do
`.createView({ dimension: "2d", baseArrayLayer })`
instead of
`.createView({ baseArrayLayer, arrayLayerCount: 1 })`

([scratch doc](https://docs.google.com/spreadsheets/d/1Y9Fmtv8ojhWB21KxBRdC6abW5QoQfQNPIOCfyzVpbG8/edit?usp=sharing))

Reply:
Resolved the other nits so we can focus on this. I'm a bit torn on this as well. It does feel like the current algorithm I describe is a bit too much magic, but it also seems like it would be annoying if, for example, I asked for a `"cube"` and it wasn't smart enough to default to an `arrayLayerCount` of `6`.

Where we draw the line in that spectrum is something I'm eager to discuss at the next editors meeting. 😃

Reply:
- If resolved.dimension is undefined:
    - If texture.dimension is 1d: set resolved.dimension to 1d.
    - If texture.dimension is 2d: set resolved.dimension to 2d.
    - If texture.dimension is 3d: set resolved.dimension to 3d.
- If resolved.arrayLayerCount is undefined:
    - If descriptor.dimension is "1d", "2d", or "3d", set resolved.arrayLayerCount to 1.
      Otherwise, set resolved.arrayLayerCount to (texture layer count - descriptor.baseArrayLayer).
      (Note this could result in an invalid descriptor if the dimension is cube or cube-array.)

Reply:
Updated the PR to include this change to the logic, but I didn't make the change @kvark requested to move the validation out into its own algorithm. I'd like to push that into its own change where I can do several related moves at the same time.

Reply:
SGTM, that's going to be more experimental anyway.